src/main/java/org/folio/fqm/service/ResultSetService.java

```diff
@@ -34,10 +30,11 @@
 @Log4j2
 public class ResultSetService {
 
-  private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-  private static final DateTimeFormatter DATE_TIME_OFFSET_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'+00:00'");
-  private static final String DATE_TIME_REGEX = "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\.\\d{3}Z$";
-  private static final String DATE_TIME_OFFSET_REGEX = "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\.\\d{3}[+-]\\d{2}:\\d{2}$";
+  private static final DateTimeFormatter DATE_TIME_FORMATTER = new DateTimeFormatterBuilder()
+    .append(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+    .optionalStart().appendOffsetId() // optional Z/timezone at end
+    .toFormatter().withZone(ZoneOffset.UTC); // force interpretation as UTC
+  private static final String DATE_TIME_REGEX = "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\.\\d{3}([+-]\\d{2}:\\d{2}(:\\d{2})?|Z|)$";
```

Comment:
bad news: technically, timezone offsets can have a seconds place.

anyways, this combines the previous regex into one, since both can get parsed by the same formatter. 
For convenience:
<img width="1312" alt="image" src="https://github.com/user-attachments/assets/6720ec6e-4552-4e20-890c-015ee69f23e8">